packages/dev/core/src/Lights/Clustered/clusteredLight.ts

```diff
@@ -0,0 +1,261 @@
+import { Constants } from "core/Engines/constants";
+import type { AbstractEngine } from "core/Engines/abstractEngine";
+import type { ThinEngine } from "core/Engines/thinEngine";
+import type { Effect } from "core/Materials/effect";
+import { RenderTargetTexture } from "core/Materials/Textures/renderTargetTexture";
+import { Vector3 } from "core/Maths/math.vector";
+import { Color4, TmpColors } from "core/Maths/math.color";
+import type { Mesh } from "core/Meshes/mesh";
+import { Logger } from "core/Misc/logger";
+import { _WarnImport } from "core/Misc/devTools";
+import type { Scene } from "core/scene";
+
+import { LightMaskMaterial } from "./lightMaskMaterial";
+import { Light } from "../light";
+import { LightConstants } from "../lightConstants";
+import { PointLight } from "../pointLight";
+import { SpotLight } from "../spotLight";
+
+export class ClusteredLight extends Light {
+    private static _GetEngineMaxLights(engine: AbstractEngine): number {
+        const caps = engine._caps;
+        if (!engine.supportsUniformBuffers || !caps.texelFetch) {
+            return 0;
+        } else if (engine.isWebGPU) {
+            // On WebGPU we use atomic writes to storage textures
+            return 32;
+        } else if (engine.version > 1) {
+            // On WebGL 2 we use additive float blending as the light mask
+            if (!caps.colorBufferFloat || !caps.blendFloat) {
+                return 0;
+            }
+            // Due to the use of floats we want to limit lights to the precision of floats
+            const gl = (<ThinEngine>engine)._gl;
```

Comment:
No gl calls should be outside of engine :-)

Reply:
Do we have some existing way to query floating point precision or do I need to add one?